routes/project/billing.rb

```diff
@@ -168,8 +168,13 @@ class Clover
 
             unless invoice.status == "current"
               response["Content-Type"] = "application/pdf"
-              response["Content-Disposition"] = "filename=\"#{@invoice_data[:filename]}.pdf\""
-              next invoice.generate_pdf(@invoice_data)
+              response["Content-Disposition"] = "inline; filename=\"#{invoice.filename}\""
+              begin
+                next Invoice.blob_storage_client.get_object(bucket: Config.invoices_bucket_name, key: invoice.blob_key).body.read
+              rescue Aws::S3::Errors::NoSuchKey
+                Clog.emit("Could not find the invoice") { {not_found_invoice: {invoice_ubid: invoice.ubid}} }
+                next invoice.generate_pdf(@invoice_data)
```

Comment:
Should we persist the generated PDF if it doesn't exist?

Reply:
We shouldn't normally hit this branch. All invoices should be persisted. I added this to prevent unhandled exceptions. I want to debug it more before persisting.